azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/AvailabilitySets.java

```diff
@@ -3,25 +3,19 @@
 import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsDeletingByGroup;
 import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsGettingByGroup;
 import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsListingByGroup;
-import com.microsoft.azure.management.resources.fluentcore.collection.SupportsCreating;
-import com.microsoft.azure.management.resources.fluentcore.collection.SupportsDeleting;
-import com.microsoft.azure.management.resources.fluentcore.collection.SupportsListing;
+import com.microsoft.azure.management.resources.fluentcore.collection.ResourcesInGroup;
 
 /**
  * Entry point to availability set management API.
  */
-public interface AvailabilitySets extends
-        SupportsListing<AvailabilitySet>,
+public interface AvailabilitySets extends 
+        ResourcesInGroup<AvailabilitySet, AvailabilitySet.DefinitionBlank>,
         SupportsListingByGroup<AvailabilitySet>,
         SupportsGettingByGroup<AvailabilitySet>,
-        SupportsCreating<AvailabilitySet.DefinitionBlank>,
-        SupportsDeleting,
         SupportsDeletingByGroup {
-    /**
-     * Entry point to availability set management API within a specific resource group.
-     */
-    interface InGroup extends
-            SupportsListing<AvailabilitySet>,
-            SupportsCreating<AvailabilitySet.DefinitionCreatable>,
-            SupportsDeleting {}
+    
+    
+    public interface InGroup 
+        extends ResourcesInGroup<AvailabilitySet, AvailabilitySet.DefinitionAfterGroup> {
```

Comment:
actually, just as I was changing this back -- the other (and actually bigger) reason I added this is because it opens up an opportunity for bugs between InGroup and the not ingroup. Note that this is also refercnes in the collection interface. If you change the order, nothing will tell you that you need to fix the InGroup interface as well. So we'll have bugs where InGroup doesn't have the same sequence as the global collection... See what I mean?  Look at AvailabilitySets interface to see what I mean.